vscode-alive.md

```diff
@@ -0,0 +1,448 @@
+           :description ""
+           :in-order-to ((test-op (load-op "experiment/test/all")))
+           :perform (test-op (o c) (symbol-call :test/all :test-suite)))
+
+         (defsystem "experiment/test"
+           :depends-on ("experiment/test/all"))
+
+         (register-system-packages "experiment/src/app" '(:app))
+         (register-system-packages "experiment/test/all" '(:test/all))
+         ~~~
+
+### Optional Custom Configurations
+
+#### Configuring VSCode Alive to work with Quicklisp
+
+Assuming that you have quicklisp installed and configured to load on
+init, quicklisp just works.
+
+#### Configuring VSCode Alive to work with CLPM
+
+Assuming that you have [CLPM](https://clpm.dev) installed and
+configured, [modify your vscode
+settings](https://code.visualstudio.com/docs/getstarted/settings) to
+look like this:
+
+1. Add the following to to your [VSCode settings](https://code.visualstudio.com/docs/getstarted/settings):
+```json
+  "alive.swank.startupCommand":[
+    "clpm",
+    "exec",
+    "--with-client",
+    "--context=$(sbcl \"--noinform\" \"--eval\" \"(require :asdf)\" \"--eval\" \"(write-string (nt(remove-if (lambda (s) (= (length s) 0)) (uiop/utility:split-string (namestring (uiop:getcwd)) :(string (uiop:directory-separator-for-host)))))))\" --eval \"(exit)\")",
+    "--",
+    "sbcl",
+    "--eval",
+    "(asdf:load-system :swank)",
+    "--eval",
+    "(swank:create-server)"
+  ],
+```
+
+_This will start up a clpm context named after the directory of the
+workspace or containing directory of the file on which you're working.
+That context line looks the way it does because it is effectively
+"scripting" in SBCL.  Instead of using bash or powerscript and leaving
+some users out, this example uses lisp to determine the folder name of
+the current working directory.  This is convenient because its
+cross-platform.  This config works on mac, linux, and windows._
+
+#### Configuring VSCode Alive to work with Roswell
+
+Assuming that you have [Roswell](https://roswell.github.io/) installed,
+[modify your vscode
+settings](https://code.visualstudio.com/docs/getstarted/settings) to
+look like this:
+
+```json
+  "alive.swank.startupCommand": [
+    "ros",
+    "run",
+    "--eval",
+    "(require :asdf)",
+    "--eval",
+    "(asdf:load-system :swank)",
+    "--eval",
+    "(swank:create-server)"
+  ]
+```
+
+#### Connecting VSCode Alive to a Docker container
+
+![](assets/vscode-gifs/docker-connect.gif)
+
+_These instructions will work for remote connections, wsl connections,
+and github Codespaces as well using the `Remote - SSH` and `Remote -
+WSL`, and `Github Codespaces` extensions, respectively assuming you have
+the extensions installed.  For this example, make sure you have the
+[Containers extension installed and
+configured](https://code.visualstudio.com/docs/remote/containers)._
+
+1. Pull a docker image that has sbcl installed, in this example, we'll use
+   the latest daewok sbcl.
```

Comment:
ah, only thing, the daewok images are deprecated, the author now maintains the clfoundation ones: https://hub.docker.com/r/clfoundation/sbcl / https://gitlab.common-lisp.net/cl-docker-images/sbcl/

Reply:
fixed.